Intro (#51)
infinum/eightshift-docs: website/docs/basics/global-settings.md

---
id: global-settings
title: Global Settings
sidebar_label: Global Settings
---

[![docs-source](https://img.shields.io/badge/source-eigthshift--frontend--libs-yellow?style=for-the-badge&logo=javascript&labelColor=2a2a2a)](https://github.com/infinum/eightshift-frontend-libs)

> Ok, this is the part that we are incredibly proud of. Did you know that you can pass values from the JSON manifest or some other file directly to SCSS? Yes, I had the same facial expression when I found out. For this, to work, we used Webpack and its excellent features.

Reviewer: ```suggestion
Ok, this is the part that we are incredibly proud of. Did you know that you can pass values from the JSON manifest or some other file directly to SCSS? Yes, we had the same facial expression when we found out. For this, to work, we used Webpack and its excellent features.
```

Maybe we should we just stick to the we instead of I.

> This feature comes only with the block setup.

> So if you used the `init_theme` setup, you are all set; if not, please visit [blocks chapter](blocks) for more details.

Reviewer 2: ```suggestion
If you used the `init_theme` setup, you are all set; if not, please visit [blocks chapter](blocks) for more details.
```

> ## Why we use this?

Reviewer 2: ```suggestion
## Why do we use this?
```

> Because with dynamic blocks, you must create your blocks, we got tired of copy/pasting styles from JavaScript/PHP/SCSS files. That is why we created this feature to define some global styles at one spot and use them in all three technologies.

Reviewer 2: ```suggestion
Because, with dynamic blocks, you must create your blocks. We got tired of copy/pasting styles from JavaScript/PHP/SCSS files. That is why we created this feature to define some global styles in one spot and use them in all three places.
```

> ## How this works?

Reviewer 2: ```suggestion
## How does this work?
```

You have file `src/Blocks/manifest.json` that has a key called `globalVariables`. We use this key and build the SCSS maps that you can use in your style files.

Mind-blowing right?

To use it, write your style in the `manifest.json` under the `globalVariables` key, and you can use that variable in your style files like this:

> ```js

Reviewer 2: ```suggestion
```scss
```

Author: I will ignore all the comments like this because scss styling in not available in docusaurus and PHP style looks a blah gray and not good at all. So let's leave this like it is

Reviewer 2: That's sad :(

Did you try adding additional languages: https://v2.docusaurus.io/docs/markdown-features/#syntax-highlighting

Odd that scss wouldn't be supported 🤷‍♂️

Author: I will check but not sure

$base-col-number: global-settings(maxCols);
```

Keep in mind that the `global-settings` is a function that checks the `globalVariables` map and returns the value depending on the key provided.

For example, if you have something like this in your `manifest.json`:

> ```js

Reviewer 2: ```suggestion
```json
```

{
  "globalVariables": {
    "maxCols": 12,
    "gutters": {
      "none": "0",
      "default": "25px",
      "big": "50px"
    },
    "colors": [
      {
        "name": "Primary",
        "slug": "primary",
        "color": "#3E3E3E"
      },
      {
        "name": "Black",
        "slug": "black",
        "color": "#000000"
      },
    ]
  }
}
```

### Scenario 1:

You type this in you SCSS files:

> ```js

Reviewer 2: ```suggestion
```scss
```

$base-col-number: global-settings(maxCols);
```

the output will be: **12**

### Scenario 2:

You type this in you SCSS files:

> ```js

Reviewer 2: ```suggestion
```scss
```

$gutters: global-settings(gutters);
```

> the output will be a map that you need to loop or fetch individual files by `get-map-strict` or `get-map-deep` function:

Reviewer 2: ```suggestion
the output will be a map that you need to loop or fetch individual files using `get-map-strict` or `get-map-deep` function:
```

> ```js

Reviewer 2: ```suggestion
```json
```

$gutters: (
  "none": "0",
  "default": "25px",
  "big": "50px"
);
```

so you can call it like this:

> ```js

Reviewer 2: ```suggestion
```scss
```

width: get-map-strict($gutters, default);
```

### Scenario 3:

I want to output colors, but they don't follow the structure you mentioned before. Why is that?

> Yes, colors are different because we use this manifest to output the colors to the Block Editors color palette, which is why colors are different. (Only colors are like this).

Reviewer 2: ```suggestion
Yes, colors are different because we use this manifest to output the colors to the Block Editors' color palette, which is why colors are different. Only colors are like this.
```

You type this in you scss files:

> ```js

Reviewer 2: ```suggestion
```scss
```

$primary-color: global-settings(colors, primary),
$black-color: global-settings(colors, black),
```

the output will be:

> ```js

Reviewer 2: ```suggestion
```scss
```

$primary-color: #3E3E3E;
$black-color: #000000;
```

so you use it the normal SCSS way:

> ```js

Reviewer 2: ```suggestion
```scss
```
